test/integration/short/client-execute-prepared-tests.js

```diff
@@ -12,8 +12,117 @@ var vit = helper.vit;
+      client.shutdown.bind(client);
+      sCluster.destroy(done);
+    });
+    it('should prepare query on all hosts', function (done) {
+      var idRandom = types.Uuid.random();
+      client.execute(query, [idRandom], {prepare: 1}, function (err, result) {
+        assert.ifError(err);
+        assert.strictEqual(client.hosts.length, 5);
+        assert.notEqual(result, null);
+        assert.notEqual(result.rows, null);
+        utils.eachSeries(client.hosts.values(), function(host, next) {
+          sCluster.queryNodeLog(host.address, function(err, log) {
+            assert.ifError(err);
+            var prepareQuery = log.find(function(queryLog) {
+              return queryLog.type.startsWith("PREPARE");
+            });
+            assert.notEqual(prepareQuery, undefined);
+            next();
+          });
+        }, done);
+      });
+    });
+    it('should re-prepare query when host go UP again', function (done) {
+      var idRandom = types.Uuid.random();
+      var nodeDownAddress = sCluster.dcs[0].nodes[4].address;
+      utils.series(
+        [
+          function stopLastNode(next) {
+            sCluster.stopNode(nodeDownAddress, next);
+          },
+          function prepareQuery(next) {
+            client.execute(query, [idRandom], {prepare: 1}, function (err, result) {
+              assert.ifError(err);
+              assert.strictEqual(client.hosts.length, 5);
+              assert.notEqual(result, null);
+              assert.notEqual(result.rows, null);
+              next();
+            });
+          },
+          function verifyLogs(next) {
+            utils.eachSeries(client.hosts.values(), function(host, nextHost) {
+              sCluster.queryNodeLog(host.address, function(err, log) {
+                assert.ifError(err);
+                var prepareQuery = log.find(function(queryLog) {
+                  return queryLog.type.startsWith("PREPARE");
+                });
+                if (prepareQuery == undefined) {
+                  assert.strictEqual(nodeDownAddress, host.address);
+                } else {
+                  assert.notEqual(prepareQuery, undefined);
+                }
+                nextHost();
+              });
+            }, next);
+          },
+          function resumeLastNode(next) {
+            var timeout = null;
+            client.hosts.get(nodeDownAddress).on('up', function() {
+              clearTimeout(timeout);
+              next();
+            });
+            sCluster.resumeNode(nodeDownAddress, function() {
+              timeout = setTimeout(next, 2000);
+            });
+          },
+          function verifyPrepareQueryOnLastNode(next) {
+            sCluster.queryNodeLog(nodeDownAddress, function(err, log) {
+              assert.ifError(err);
+              var prepareQuery = log.find(function(queryLog) {
+                return queryLog.query.startsWith("PREPARE");
+              });
+              assert.notEqual(prepareQuery, undefined);
```

Comment:
This tests is failing here. This tests the ability of the driver to prepare a statement in a node recently re-started. Maybe the test is not reflecting this scenario properly.

Reply:
Check whether the node is being marked as back up.